src/main/java/spoon/reflect/declaration/CtModule.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (C) 2006-2017 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.reflect.declaration;
+
+import spoon.reflect.annotations.PropertyGetter;
+import spoon.reflect.annotations.PropertySetter;
+import spoon.reflect.reference.CtModuleReference;
+import spoon.support.DerivedProperty;
+
+import java.util.List;
+
+import static spoon.reflect.path.CtRole.EXPORTED_PACKAGE;
+import static spoon.reflect.path.CtRole.MODIFIER;
+import static spoon.reflect.path.CtRole.MODULE_MEMBER;
+import static spoon.reflect.path.CtRole.OPENED_PACKAGE;
+import static spoon.reflect.path.CtRole.PROVIDED_SERVICE;
+import static spoon.reflect.path.CtRole.REQUIRED_MODULE;
+import static spoon.reflect.path.CtRole.SERVICE_TYPE;
+import static spoon.reflect.path.CtRole.SUB_PACKAGE;
+
+/**
+ * Represents a Java module as defined in Java 9.
+ *
+ * Modules are defined in `module-info.java` as follows:
+ * <pre>
+ *     module com.example.foo {
+ *
+ *     }
+ * </pre>
+ *
+ * Modules define required modules, and exported packages for client code.
+ *
+ * A module can export a service, defined as a type.
+ * Provided services are implementations of given service.
+ * Modules can require services ("uses" in Java 9)
+ *
+ * An open module is a module where all packages are exported.
+ */
+public interface CtModule extends CtNamedElement {
+
+	/**
+	 * The name for the top level module.
+	 */
+	String TOP_LEVEL_MODULE_NAME = "unnamed module";
+
+	/**
+	 * Returns the unnamed module
+	 * Warning: if there are other modules, they are not contained in that unnamed root module.
```

Comment:
I do not understand this comment ... the method returns `boolean` ...

Reply:
Yeah it was a confusion between CtModel and CtModule...